### PR TITLE
Added 'update' function for softmax

### PR DESCRIPTION
**Pull request by fufjvnvnf**
I did not realized the use of such 'update' functions in some of the layers until recently, so I append the function and check for 'Softmax' modeling over the one in 'Sigmoid'. I wonder if one is also necessary for 'CrossEntropy', but another loss function, `SquareLoss` does not implement such `update` function. Happy to commit further for adding either test for `Softmax`'s case for update or `update` functions for both loss functions.

### CODE REVIEW

**Comment by stevenygd** [comment]
It's very scare to think about that the Softmax layer still works not having this function. Maybe you haven't used any of those reallocated memory, since the reason I need this function is to reallocate memory based on the change of batchsize (since we couldn't know the batchsize at the very very beginning). I will merge this PR since it looks alright to me, but you will make an issue to check the allocation problem for SoftMaxLayer? Thanks!
